code/game/objects/structures.dm

```diff
@@ -64,26 +64,26 @@
 	if(C == user)
 		do_climb(user)
 
+/obj/structure/proc/density_check()
+	for(var/obj/O in range(0, src))
+		if(O.density == 1 && O != src && !istype(O, /obj/machinery/door/window)) //Ignores windoors, as those already block climbing, otherwise a windoor on the opposite side of a table would prevent climbing.

```

Comment:
Only need to check `if(O.density`, not `if(O.density == 1`

Reply:
Additionally, I can't pull up the docs right now, but I think `orange` will remove `src` from the list automatically.